<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenPoster</class>
 <widget class="QMainWindow" name="OpenPoster">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1280</width>
    <height>800</height>
   </size>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern"> 
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="0">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string notr="true">OpenPoster</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    color: palette(text);
    background-color: transparent;
	spacing: 0px;
}

QWidget:focus {
    outline: none;
}

QWidget [cls=central] {
    background-color: palette(base);
	border-radius: 0px;
	border: 1px solid palette(mid);
}

QLabel {
    font-size: 14px;
}

QToolButton {
    background-color: palette(button);
    border: none;
    color: palette(text);
    font-size: 14px;
	min-height: 35px;
	icon-size: 16px;
	padding-left: 10px;
	padding-right: 10px;
	border-radius: 8px;
}

QToolButton[cls=sidebarBtn] {
    background-color: transparent;
	icon-size: 24px;
}

QToolButton:pressed {
    background-color: palette(dark);
    color: palette(text);
}

QToolButton:checked {
    background-color: palette(highlight);
    color: palette(highlighted-text);
}

QCheckBox {
	spacing: 8px;
	font-size: 14px;
}

QRadioButton {
	spacing: 8px;
	font-size: 14px;
}

QLineEdit {
	border: none;
	background-color: transparent;
	color: palette(text);
	font-size: 14px;
}

QScrollBar:vertical {
    background: transparent;
    width: 8px;
}

QScrollBar:horizontal {
    background: transparent;
    height: 8px;
}

QScrollBar::handle {
    background: palette(mid);
    border-radius: 4px;
}

QScrollBar::handle:pressed {
    background: palette(dark);
}

QScrollBar::add-line,
QScrollBar::sub-line {
    background: none;
}

QScrollBar::add-page,
QScrollBar::sub-page {
    background: none;
}

QSlider::groove:horizontal {
    background-color: palette(mid);
    height: 4px;
	border-radius: 2px;
}

QSlider::handle:horizontal {
    background-color: palette(dark);
    width: 8px;
    margin: -8px 0;
    border-radius: 4px;
}

QSlider::handle:horizontal:pressed {
    background-color: palette(highlight);
}

QSlider::tick:horizontal {
    background-color: palette(dark);
    width: 1px;
}

QHeaderView::section {
    background-color: palette(button);
    color: palette(text);
    padding: 8px;
    border: none;
    border-right: 1px solid rgba(120, 120, 120, 60);
    border-bottom: none;
    height: 30px;
}

QTableView, QTreeView {
    border: none;
    selection-background-color: palette(highlight);
    selection-color: palette(highlighted-text);
}

QTreeView::item {
    min-height: 30px;
    padding: 2px;
}

QPushButton, QLabel[clickable="true"] {
    border-radius: 6px;
    padding: 5px 10px;
}

QLabel[objectName="previewLabel"] {
    font-size: 14px; 
    padding: 5px;
}

QPushButton[objectName="editButton"] {
    border: none; 
    background-color: transparent;
}

QPushButton[objectName="editButton"]:hover {
    background-color: rgba(128, 128, 128, 30);
    border-radius: 20px;
}

QPushButton[objectName="editButton"]:checked {
    background-color: rgba(0, 120, 215, 50);
    border-radius: 20px;
}

QPushButton[objectName="playButton"] {
    border: none; 
    background-color: transparent;
}

QPushButton[objectName="playButton"]:hover {
    background-color: rgba(128, 128, 128, 50);
    border-radius: 20px;
}

QPushButton[objectName="settingsButton"] {
    border: none; 
    background-color: transparent;
}

QPushButton[objectName="settingsButton"]:hover {
    background-color: rgba(128, 128, 128, 30);
    border-radius: 20px;
}

QPushButton[objectName="exportButton"] {
    border: 1px solid gray;
    border-radius: 8px;
    padding: 5px 10px;
}

QPushButton[objectName="exportButton"]:hover {
    background-color: rgba(128, 128, 128, 30);
}

QTableWidget {
    border: none;
    background-color: transparent;
    gridline-color: transparent;
}

QTableWidget::item { 
    padding: 8px;
    min-height: 30px;
}

QTableWidget::item:first-column {
    border-right: 1px solid rgba(120, 120, 120, 60);
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="headerWidget" native="true">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>40</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_header">
       <property name="leftMargin">
        <number>5</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
       <item>
        <widget class="QPushButton" name="openFile">
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton {
  border: 1.5px solid palette(highlight);
  border-radius: 8px;
  padding: 3px 8px;
  background-color: rgba(80, 120, 200, 30);
}
QPushButton:pressed {
  background-color: rgba(60, 100, 180, 120);
}</string>
         </property>
         <property name="text">
          <string>Open File</string>
         </property>
         <property name="fixedHeight">
          <number>30</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="filename">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>28</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 1.5px solid palette(highlight); border-radius: 8px; padding: 3px 8px; color: #666666; font-style: italic;</string>
         </property>
         <property name="text">
          <string>No File Open</string>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="fixedHeight">
          <number>30</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="exportButton">
         <property name="text">
          <string>Export</string>
         </property>
         <property name="toolTip">
          <string>Export</string>
         </property>
         <property name="iconSize">
          <size>
           <width>18</width>
           <height>18</height>
          </size>
         </property>
         <property name="fixedHeight">
          <number>30</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="settingsButton">
         <property name="toolTip">
          <string>Settings</string>
         </property>
         <property name="fixedSize">
          <size>
           <width>30</width>
           <height>30</height>
          </size>
         </property>
         <property name="iconSize">
          <size>
           <width>18</width>
           <height>18</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="layersWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 1px solid #4B4B4B; border-radius: 4px; padding: 4px;</string>
       </property>
       <layout class="QVBoxLayout" name="layersLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QSplitter" name="layersSplitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="QWidget" name="layersSection" native="true">
           <layout class="QVBoxLayout" name="layersSectionLayout">
            <item>
             <widget class="QLabel" name="layersLabel">
              <property name="text">
               <string>Layers</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTreeWidget" name="treeWidget">
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="columnCount">
               <number>3</number>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="indentation">
               <number>20</number>
              </property>
              <property name="uniformRowHeights">
               <bool>true</bool>
              </property>
              <attribute name="headerVisible">
               <bool>true</bool>
              </attribute>
              <attribute name="headerDefaultSectionSize">
               <number>200</number>
              </attribute>
              <attribute name="headerMinimumSectionSize">
               <number>100</number>
              </attribute>
              <attribute name="headerHighlightSections">
               <bool>true</bool>
              </attribute>
              <attribute name="headerStretchLastSection">
               <bool>true</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Type</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>ID</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="statesSection" native="true">
           <layout class="QVBoxLayout" name="statesSectionLayout">
            <item>
             <widget class="QLabel" name="statesLabel">
              <property name="text">
               <string>States</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTreeWidget" name="statesTreeWidget">
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="columnCount">
               <number>3</number>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
              <property name="indentation">
               <number>20</number>
              </property>
              <property name="uniformRowHeights">
               <bool>true</bool>
              </property>
              <attribute name="headerVisible">
               <bool>true</bool>
              </attribute>
              <attribute name="headerDefaultSectionSize">
               <number>200</number>
              </attribute>
              <attribute name="headerMinimumSectionSize">
               <number>100</number>
              </attribute>
              <attribute name="headerHighlightSections">
               <bool>true</bool>
              </attribute>
              <attribute name="headerStretchLastSection">
               <bool>true</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Type</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Value</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="previewWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>0</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 1px solid #4B4B4B; border-radius: 4px; padding: 4px;</string>
       </property>
       <layout class="QVBoxLayout" name="previewLayout">
        <item>
         <layout class="QHBoxLayout" name="previewHeaderLayout_ui">
          <item>
           <widget class="QLabel" name="previewLabel">
            <property name="text">
             <string>Preview</string>
            </property>
            <property name="styleSheet">
             <string>font-size: 14px; padding: 5px;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="previewHeaderSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="editButton">
            <property name="toolTip">
             <string>Toggle Edit Mode</string>
            </property>
            <property name="fixedSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="playButton">
            <property name="toolTip">
             <string>Play/Pause Animations</string>
            </property>
            <property name="fixedSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="iconSize">
             <size>
              <width>32</width>
              <height>32</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="CustomGraphicsView" name="graphicsView">
          <property name="objectName">
           <string notr="true">graphicsView</string>
          </property>
          <property name="cursor">
           <cursorShape>OpenHandCursor</cursorShape>
          </property>
          <property name="renderHints">
           <set>QPainter::Antialiasing|QPainter::SmoothPixmapTransform</set>
          </property>
          <property name="transformationAnchor">
           <enum>QGraphicsView::AnchorUnderMouse</enum>
          </property>
          <property name="resizeAnchor">
           <enum>QGraphicsView::AnchorUnderMouse</enum>
          </property>
          <property name="viewportUpdateMode">
           <enum>QGraphicsView::FullViewportUpdate</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="inspectorWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 1px solid #4B4B4B; border-radius: 4px; padding: 4px;</string>
       </property>
       <layout class="QVBoxLayout" name="inspectorLayout">
        <item>
         <widget class="QLabel" name="inspectorLabel">
          <property name="text">
           <string>Inspector</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="tableWidget">
          <property name="columnCount">
           <number>2</number>
          </property>
          <property name="showGrid">
           <bool>false</bool>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>100</number>
          </attribute>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>120</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderSectionResizeMode">
           <enum>QHeaderView::Interactive</enum>
          </attribute>
          <attribute name="verticalHeaderMinimumSectionSize">
           <number>25</number>
          </attribute>
          <attribute name="verticalHeaderDefaultSectionSize">
           <number>30</number>
          </attribute>
          <column>
           <property name="text">
            <string>Key</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Value</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>gui.custom_widgets</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
